### PR TITLE
more work on the BN/VC split

### PR DESCRIPTION
**Pull request by onqtam**
- code cleanup, better logging & retrying requests on the VC side if the BN fails for some reason (using exceptions for errors when a request fails or cannot be served)
- updated the validator shell script after the keystore changes
- better naming for consoles when using multitail

### CODE REVIEW

File: beacon_chain/validator_api.nim [hunk 1]
```diff
@@ -50,102 +51,93 @@ proc installValidatorApiHandlers*(rpcServer: RpcServer, node: BeaconNode) =
...
-    # TODO how do we handle the case when we cannot return a meaningful block? 404...
-    doAssert(res.message.isSome())
-    return res.message.get(BeaconBlock()) # returning a default if empty
+    if res.message.isNone():
+      raise newException(CatchableError, "could not retrieve block for slot: " & $slot)
+    return res.message.get()
 
   rpcServer.rpc("post_v1_beacon_blocks") do (body: SignedBeaconBlock) -> bool:
-    notice "== post_v1_beacon_blocks"
+    debug "post_v1_beacon_blocks", slot = body.message.slot, prop_idx = body.message.proposer_index
```
Comment (arnetheduck) [inline beacon_chain/validator_api.nim:80]: 80 char line length

File: beacon_chain/validator_client.nim [hunk 2]
```diff
@@ -43,6 +44,27 @@ type
...
+      error "Could not connect to the BN - retrying!", err = err.msg
+      await sleepAsync(chronos.seconds(1)) # 1 second before retrying
+
+template attemptUntilSuccess(vc: ValidatorClient, body: untyped) =
+  while true:
+    try:
+      body
+      break
+    except CatchableError as err:
+      error "Caught an unexpected error", err = err.msg
```
Comment (arnetheduck) [inline beacon_chain/validator_client.nim:65]: `error` is for unrecoverable errors where the application cannot continue

Comment (arnetheduck) [reply beacon_chain/validator_client.nim:65]: `notice` is or `info` are less alarmist here

**Comment by onqtam** [comment]
pushing the changes from the CLI

**Pull request closed without merging**